tests/common/schema/test_references.py

```diff
@@ -0,0 +1,437 @@
+import pytest
+
+import dlt
+from dlt.common.schema.utils import (
+    create_root_child_reference,
+    get_all_root_child_references_from_root,
+    create_parent_child_reference,
+    get_all_parent_child_references_from_root,
+    create_load_table_reference,
+    create_version_and_loads_hash_reference,
+    create_version_and_loads_schema_name_reference,
+)
+
+from tests.utils import preserve_environ, patch_home_dir, autouse_test_storage
+
+
+@pytest.fixture
+def schema():
+    return dlt.Schema.from_dict(
```

Comment:
up to you, but I save those as yaml files for example
```py
with open("tests/common/cases/schemas/eth/ethereum_schema_v11.yml", "r", encoding="utf-8") as f:
        schema = dlt.Schema.from_dict(yaml.safe_load(f))
```

Reply:
end to end test is missing: schema.references(self) should produce all expected references that you enumerate below

Reply:
to weed off naming convention problems: parametrize your test with naming convention. one that certainly produces mixed case identifiers:

```py
@pytest.fixture
def ci_gcp_client(empty_schema: Schema) -> BigQueryClient:
    empty_schema._normalizers_config["names"] = "tests.common.cases.normalizers.title_case"
    empty_schema.update_normalizers()
```